__tests__/UserCaption-test.js

```diff
@@ -1,12 +1,186 @@
 import 'react-native';
 import React from 'react';
 import UserCaption from '@screens/app/Components/UserCaption/UserCaption';
+import checkPropTypes from 'check-prop-types';
+var assert = require('assert');
+import {shallow} from 'enzyme';
 
 import renderer from 'react-test-renderer';
 
 test('user caption snapshot test', () => {
-  const tree = renderer.create(<UserCaption />).toJSON();
+  const tree = renderer
+    .create(<UserCaption username="test" caption="test" />)
+    .toJSON();
   expect(tree).toMatchSnapshot();
 });
+// test numbers, arrays, strings, booleans combinations of other data types
+describe('testing UserCaption values, should not throw errors', () => {
+  const stringCaptionTest = 'Hi World, Missed Me ?';
+  const stringUsernameTest = 'e805';
+  it('testing UserCaption Creation String', () => {
+    const wrapper = shallow(
+      <UserCaption username={stringUsernameTest} caption={stringCaptionTest} />,
+    );
+    const instance = wrapper.instance();
+    const currentUsername = instance.state.username;
+    const currentCaption = instance.state.caption;
```

Comment:
suggest use es6 syntax where you can do const {username, caption } = instance.state to reduce redundnacy

Reply:
Created Function, so this no longer applies